src/lib/repositories_v2/impl/chef_repositoty_impl.py

```diff
@@ -0,0 +1,67 @@
+from datetime import datetime
+
+from src.constants.audit import Status
+from src.core.ioc import get_ioc_instance
+from src.lib.entities.sqlalchemy_orm_mapping import Chef, Order
+from src.lib.repositories_v2.chef_repository import ChefRepository
+
+
+class ChefRepositoryImpl(ChefRepository):
+    def __init__(self):
+
+        ioc = get_ioc_instance()
+        self.sqlalchemy_session = ioc.get_instance("orm_session")
+
+    def add(self, chef):
+        chef.created_date = datetime.now()
+        chef.updated_by = chef.created_by
+        chef.updated_date = chef.created_date
+        self.sqlalchemy_session.add(chef)
+        self.sqlalchemy_session.commit()
+
+    def get_by_id(self, chef_id):
+        return (
+            self.sqlalchemy_session.query(Chef)
+            .filter(Chef.id == chef_id)
+            .filter(Chef.entity_status == Status.ACTIVE.value)
+            .first()
+        )
+
+    def get_all(self):
+        chefs = self.sqlalchemy_session.query(Chef).filter(
+            Chef.entity_status == Status.ACTIVE.value
+        )
+        return chefs
+
+    def delete_by_id(self, chef_id, chef):
+        chef_to_be_delete = (
+            self.sqlalchemy_session.query(Chef).filter(Chef.id == chef_id).first()
+        )
+        chef_to_be_delete.entity_status = Status.DELETED.value
+        chef_to_be_delete.updated_date = datetime.now()
+        chef_to_be_delete.updated_by = chef.updated_by
+        self.sqlalchemy_session.add(chef_to_be_delete)
+        self.sqlalchemy_session.commit()
+
+    def update_by_id(self, chef_id, chef):
+        chef_to_be_updated = (
+            self.sqlalchemy_session.query(Chef).filter(Chef.id == chef_id).first()
+        )
+        chef_to_be_updated.name = chef.name or chef_to_be_updated.name
+        chef_to_be_updated.description = (
+            chef.description or chef_to_be_updated.description
+        )
+        chef_to_be_updated.updated_date = datetime.now()
+        chef_to_be_updated.updated_by = chef.updated_by
+        self.sqlalchemy_session.add(chef_to_be_updated)
+        self.sqlalchemy_session.commit()
+
+    def get_available_chefs(self):
+        available_chef_ids = self.sqlalchemy_session.query(Chef.id).filter(
+            Chef.id.not_in(
```

Comment:
I think`not exists` is more efficient here since this query needs to do more operations.

This query:

1. Needs to find the None chef ids in order which may have a high number of records
2. Per each record within chef it compares all `None` records orders.

Step 2 does not make sense at all. You could just check if a chef id exists for all orders different from entity status active, and not in order status (new, completed, cancel)

Reply:
Also you need to check for active chefs

Reply:
done.